### PR TITLE
App Testing

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
First step in making this more broadly useful.

This PR:
* Fixes an issue with multi-select case lists 
* Makes the UI to a domain specific UI instead of admin only (gated by the APP_TESTING feature flag).

Future changes:
* translate strings
* add a DSL to make it easier to author workflows
* 'expect' steps for testing state
* option run tests on app build?

## Technical Summary
Make views domain specific & add new steps for multi-select case lists

## Feature Flag
APP_TESTING

### Safety story
Changes only impact the app testing feature.

### Automated test coverage
Added more in this PR

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by MartinRiese** [comment]
got the following error when trying out the feature locally.
```
Request Method: 	POST
Request URL: 	http://localhost:8000/a/case-lists/apps/testing/new/
Django Version: 	4.2.11
Exception Type: 	AttributeError
Exception Value: 	

'WebUser' object has no attribute 'domain'

Exception Location: 	/Users/martinriese/git/commcare-hq/corehq/apps/users/models.py, line 1678, in __getattr__
Raised during: 	corehq.apps.app_execution.views.new_workflow
```

File: corehq/apps/app_execution/forms.py [hunk 1]
```diff
@@ -39,15 +40,25 @@ def __init__(self, *args, **kwargs):
...
+                    "domain",
+                    "app_id",
+                    "username",
+                    "sync_before_run",
+                    "form_mode",
+                    "run_every",
+                    "notification_emails",
+                    css_class="col",
+                ),
+                crispy.Div(
```
Comment (MartinRiese) [inline corehq/apps/app_execution/forms.py:60]: If I understand the form correctly, and I am not sure I do, is the workflow field an alternative to the har import? This is not clear from this layout.

File: corehq/apps/app_execution/forms.py [hunk 2]
```diff
@@ -60,13 +76,8 @@ def __init__(self, *args, **kwargs):
...
-        self.final_clean_app_id()
-        self.final_clean_username()
-        return self.cleaned_data
-
-    def final_clean_username(self):
-        domain = self.cleaned_data.get("domain")
+    def clean_username(self):
+        domain = self.request.domain
         username = self.cleaned_data.get("username")
         if username and "@" not in username:
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/app_execution/forms.py:82]: Thanks, not sure how I missed that.

File: corehq/apps/app_execution/forms.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_execution/forms.py:60]: The HAR file import is a separate input and button at the bottom of the form. Importing the file will submit the whole form, process the file and then return the user to the same page with all the existing data except the workflow is replaced by the one generated from the HAR file.

image

File: corehq/toggles/__init__.py [hunk 3]
```diff
@@ -2859,3 +2859,11 @@ def domain_has_privilege_from_toggle(privilege_slug, domain):
     namespaces=[NAMESPACE_DOMAIN],
     description='Restrict data source rebuilt from UI if the relevant data for the data source crosses a threshold'
 )
+
+APP_TESTING = StaticToggle(
+    slug='app_testing',
+    label='App Testing feature',
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/toggles/__init__.py]: 39eae9810dd1d17d49d7c2676c775137b7062583

File: corehq/apps/app_execution/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/app_execution/forms.py]: 39eae9810dd1d17d49d7c2676c775137b7062583

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**